benchmarks/benchmark_morphology.py

```diff
@@ -36,3 +37,14 @@ def peakmem_reference(self, *args):
     def peakmem_watershed(self, seed_count, connectivity, compactness):
         morphology.watershed(self.image, seed_count, connectivity,
                              compactness=compactness)
+
+
+class WatershedParallel(object):
+
+    def setup(self):
+        image = filters.sobel(data.coins())
+        self.images = ((image, 100) for _ in range(4))
```

Comment:
I'm not happy with the adaption of the informal benchmark to a smaller runtime. For the current configuration the GIL-releasing version is faster but only by a factor of 0.86. Increasing the number of "jobs" leads to the GIL-releasing version unexpectedly being slower! The same thing happens if I change the pool size meaning the current pool and job size of 4 seems to be a small sweet spot were the GIL-releasing version is slower.

Would be interesting if this only happens on my local hardware or if this behavior is universal. Right now I have a rather low confidence in the results of this particular benchmark...